Fix: Added extends value in initializer for react

yacmeno opened the pull request:
<!--
    ESLint adheres to the [JS Foundation Code of Conduct](https://js.foundation/community/code-of-conduct).
-->

**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[X] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

<!--
    If the item you've checked above has a template, please paste the template questions below and answer them. (If this pull request is addressing an issue, you can just paste a link to the issue here instead.)
-->
**Tell us about your environment**

* **ESLint Version: ^6.4.0**
* **Node Version: v11.15.0**
* **npm Version: v11.15.0**

**What parser (default, Babel-ESLint, etc.) are you using?**
I have reproduced this bug with babel-eslint and @typescript-eslint/parser
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

<!-- Paste your configuration below: -->
```js
module.exports = {
    "env": {
        "browser": true,
        "es6": true
    },
    "extends": [
        "eslint:recommended",
    ],
    "globals": {
        "Atomics": "readonly",
        "SharedArrayBuffer": "readonly"
    },
    "parser": "babel-eslint",
    "parserOptions": {
        "ecmaFeatures": {
            "jsx": true
        },
        "ecmaVersion": 2018,
        "sourceType": "module"
    },
    "plugins": [
        "react"
    ],
    "rules": {
    },
};
```

</details>

**What did you do? Please include the actual source code causing the issue.**
Executed the initializer to create a config for react apps ```./node_modules/.bin/eslint --init```. You can see the reproduced bug and its fix here: https://github.com/yacmeno/eslint-react-plugin-issue (master branch is with the upstream eslint and the fix branch with the fork)

**What did you expect to happen?**
To have the downloaded plugin be configured in a working state.

**What actually happened? Please include the actual, raw output from ESLint.****
The generated config was missing an ```extends``` value that is needed (see https://www.npmjs.com/package/eslint-plugin-react)  

<!--
    Please ensure your pull request is ready:

    - Read the pull request guide (https://eslint.org/docs/developer-guide/contributing/pull-requests)
    - Include tests for this change
    - Update documentation for this change (if appropriate)
-->

<!--
    The following is required for all pull requests:
-->

**What changes did you make? (Give an overview)**
Pushed the necessary ```extends``` option in the initializer script.

**Is there anything you'd like reviewers to focus on?**

kaicataldo commented on lib/init/config-initializer.js:
```diff
@@ -292,6 +292,7 @@ function processAnswers(answers) {
+        config.extends.push("plugin:react/recommended");
     } else if (answers.framework === "vue") {
         config.plugins = ["vue"];
         config.extends.push("plugin:vue/essential");
```
Question for the team: if we decide to do this,  do we also want to use the recommended config for the Vue plugin?

yacmeno commented:
I added an assert for the extends value in an already existing test, and also added a new test for Typescript+React initialization. I would also add the recommended react-hooks plugin: https://www.npmjs.com/package/eslint-plugin-react-hooks#installation but I'd like to confirm with you if you would also add that. It might also be a PR of its own

ilyavolodin commented:
I think I would want to avoid react hooks plugin. I really want to limit plugins that we "recommend" as to not cause issues in the future. Also react hooks are only applicable to new repos, as it's a pretty new feature that's probably not used currently in vast majority of projects.

kaicataldo commented:
Agree with @ilyavolodin, I think we should not include the React hooks plugin for now. We should definitely keep it in mind for the future.

kaicataldo commented:
And to follow up on the React hooks discussion, it looks like it's actually included as a peer dep by `eslint-plugin-react` and is now automatically installed.

kaicataldo commented:
Ah, actually, it looks like this has already been fixed in another PR. Thank you for contributing, regardless!
